Library/Homebrew/compat/tap.rb

```diff
@@ -0,0 +1,38 @@
+module CaskTapMigration
+  def initialize(user, repo)
+    super
+
+    return unless user == "caskroom"
+
+    # TODO: Remove this check after migration.
+    return unless repo == "tap-migration-test"
+
+    new_user = "Homebrew"
+    new_repo = (repo == "cask") ? repo : "cask-#{repo}"
+
+    old_name = name
+    old_path = path
+    old_remote = path.git_origin
+
+    super(new_user, new_repo)
+
+    return unless old_path.git?
+
+    new_name = name
+    new_path = path
+    new_remote = default_remote
+
+    ohai "Migrating Tap #{old_name} to #{new_name} …"
+
+    puts "Moving #{old_path} to #{new_path} …"
+    path.dirname.mkpath
+    FileUtils.mv old_path, new_path
+
+    puts "Changing remote from #{old_remote} to #{new_remote} …"
+    new_path.git_origin = new_remote
+  end
+end
+
+class Tap
+  prepend CaskTapMigration
```

Comment:
This isn’t how we generally do compat classes; might be worth a PR to update the others to this format if it’s more desirable?